src/transformers/onnx/utils.py

```diff
@@ -61,3 +64,32 @@ def compute_serialized_parameters_size(num_parameters: int, dtype: ParameterForm
         Size (in byte) taken to save all the parameters
     """
     return num_parameters * dtype.size
+
+
+def get_preprocessors(model_name: str) -> List[Union[AutoTokenizer, AutoFeatureExtractor, AutoProcessor]]:
```

Comment:
If I understand correctly, it is possible to return multiple processors for models like wav2vec, where all of the following are possible:

```python
from transformers import AutoFeatureExtractor, AutoProcessor, AutoTokenizer

model_id = "facebook/wav2vec2-base-960h"
tok = AutoTokenizer.from_pretrained(model_id)
feat = AutoFeatureExtractor.from_pretrained(model_id)
proc = AutoProcessor.from_pretrained(model_id)
```

So the current logic is to first try to automatically load the appropriate preprocessor, but if multiple are returned then the user specifies which one in the CLI right?

Reply:
Yes, exactly 😊

Reply:
Thanks for the input @patrickvonplaten ! 

The [current logic](https://github.com/huggingface/transformers/blob/d156898f3b9b2c990e5963f5030a7143d57921a2/src/transformers/onnx/__main__.py#L57) in `transformers.onnx` (on `main`, not this PR) involves checking if `config.model_type` exists in one of  `PROCESSOR_MAPPING_NAMES`, `TOKENIZER_MAPPING_NAMES` or `FEATURE_EXTRACTOR_MAPPING_NAMES`.

This fails for checkpoints like [`superb/wav2vec2-base-superb-ks`](https://huggingface.co/superb/wav2vec2-base-superb-ks) (because it tries to load the tokenizer before the feature extractor), but what we could do is use a try/except block and then get the user to specify which preprocessor to use in the CLI when an exception is raised.

Reply:
I think in the case of `Perceiver`, the problem is that the model has several modalities so we don't know which `processor` to use automatically (some of the models head use the tokenizer, others use the feature extractor).

Reply:
Thanks for your input, @patrickvonplaten, @sgugger and @lewtun! 

I adapted the logic now as you've suggested, @patrickvonplaten: If a processor exists, it is returned. Otherwise, if a tokenizer or a feature extractor exists, it is returned. If both a tokenizer and a feature extractor exist, an error is raised. The function returns `None` if no preprocessor is found. The error handling is done in the function, so calling it is more convenient.

I've left the option for the user to specify the processor in there for now. It's probably rarely used but might prove helpful in future. Do you think I should keep it or remove it?